### PR TITLE
Improve docs for Delay

### PR DESCRIPTION
**Pull request by dbrgn**
See diff.

screenshot-20220919-234000

### CODE REVIEW

**Comment by dbrgn** [comment]
Do I really have to update the CHANGELOG for a minor documentation update? Is there a way to skip this check?

File: src/timer/delay.rs [hunk 1]
```diff
@@ -58,7 +58,29 @@ impl SysDelay {
...
+///
+/// With millisecond precision, you can wait from 2 ms to 49 days.
+///
+/// ### Example: Microsecond precision
+///
+/// To instantiate this with the TIM5 timer and microsecond precision:
+///
+/// ```rust
+/// let delay = dp.TIM5.delay_us(&clocks);
+/// delay.delay_us(30_u32);
```
Comment (burrbull) [inline src/timer/delay.rs:80]:
```suggestion
/// let delay = dp.TIM5.delay_us(&clocks);
/// delay.delay_us(30_u32);
/// delay.delay_ms(5_u32);
/// delay.delay(5.millis());
/// delay.delay(3.secs());
```
From your docs someone can infer that only `delay_us` works with microsecond timer. This is not true, see suggestion.

Also you can initialize timer with custom precision to change minimum/maximum precision, not only 1 microsecond or 1 millisecond per clock.
For example 2kHz (500 us/clock) timer can wait from 1 ms to 24 days.
```rust
/// let delay = dp.TIM5.delay::<2_000>(&clocks);
/// delay.delay_ms(1_u32);
/// delay.delay_ms(320_u32);
```

**Review by burrbull — APPROVED** [review]

**Comment by burrbull** [comment]
I've added `no changelog` label, but looks like it doesn't work.
Make change in changelog, please. (There are several broken links if you don't want to show this PR)

**Review by burrbull — APPROVED** [review]

**Pull request merged**